### DISCUSSION TITLE
Request: Add Advanced Diagnostics for BMS

### DISCUSSION THREAD

**Discussion by Ne0-Hack3r** [vwt12eh8/hassio-ecoflow#33] [Ideas]
Thank you so much for this custom integration. It works very well with HA and my Delta Pro units.

I was wondering if there are any plans to expose additional sensors from the BMS in DP or EB units such as battery pack voltage/current and/or individual cell voltages (if available)?

**Comment by vwt12eh8** [comment]
We were planning to respond to your request.

If you list the entities you need here, we will consider it.

**Comment by Ne0-Hack3r** [comment]
> We were planning to respond to your request.
> If you list the entities you need here, we will consider it.

Here are the detailed sensors I believe it would be nice to have for advanced battery diagnostics...

battery_level
battery_voltage
battery_current
battery_temp
battery_cycles
battery_soh
battery_voltage_max
battery_voltage_min
battery_temp_max
battery_temp_min
battery_mos_temp_max
battery_mos_temp_min
battery_fault
battery_in_power
battery_out_power

Obviously it would be good to have this information for the entire battery bank but it if is possible to extract the data and have sensors for each individual battery cell that would be very good. This would help in determining if there is an issue with a single cell and also to chart the health and behavior of the cells and the entire battery pack over time.

Thank you for considering this enhancement.

**Comment by vwt12eh8** [comment]
Certainly the usage you suggest would be very useful in some cases.

However, there is one problem. The maintenance cost would be much higher.

It would take a lot of effort to make sure everything is correct, as the units may differ from model to model, and you would have to add a large number of entities and keep maintaining them over time.

We'll be ready to implement these things, but I think it's best to wait until we get more people's opinions before deciding whether to release it or not.

**Comment by vwt12eh8** [comment]
I will move this Issue to Discussions once. If we get a certain number of votes there, we'll finalize the release.

**Comment by vwt12eh8** [comment]
Also, I think it should be implemented after #32 is completed, because it will add a lot of entities. This will be implemented in the next release (Ver. 3.0), so it shouldn't be too far away.

**Reply by Ne0-Hack3r** [reply]
That makes sense. Thanks.
